Refactoring

fogti opened the pull request:
https://twitter.com/1kescher/status/1343964329952108546
unspaghetti'ed

actually, this PR modifies the code behavoir a bit, e.g. if a song isn't downloaded, because the destination file already exists, some debug output is skipped (e.g. moved after the condition), as this allows more efficient processing (but only works if the info wouldn't be interesting, anyways).

It is probably a good idea to squash this before merging (I think GitHub has this as an option).

fherrera124 reviewed (commented):
Since commit 011a1105911511620f7445467bb1afadc264e314, it stopped working

fogti commented:
Ah ok, I did expect something like that, esp. as I wasn't really sure what *that* part of code was doing... I'll revert that part.
ok, fixed.

fherrera124 commented:
@zserik I added your commits to my fork and made some changes and added support to retrieve the covers, can you review it? If it's worth it, you can add some changes to this PR.
876b245b3fc87159806fd6a120667f8e449a051e
PS: I'm new to Rust, so if I made a change that I shouldn't have done, feel free to point me out.

fogti committed:
Refactoring

shuffle some stmts around
move utility fns to the bottom
refac: move fname calculation into 'write_to_disk'
reorder code to improve clarity
get rid of duplicated file existence check
print invalid spotify types
handle_entry: get rid of unused generic lifetime params
cache Show and Album.name

fogti committed:
improve control flow, use enum for track vs. episode

fogti committed:
sort dependency entries in Cargo.toml alphabetically

fogti commented:
I condensed the code into 3 commits, which should still capture most of the intent behind the changes. (I especially wanted to get rid of some back-and-forth stuff and couldn't move all commits into place without introducing conflicts while rebasing, so I squashed the long run).
